### PR TITLE
[Feature #160215770] Add a post answer endpoint

### PR DESCRIPTION
**Pull request by maxwellgithinji**
**What is this PR supposed to do?**
Create a User endpoint to `POST` an answer to a question including  tests

**Descriptions of tasks to be completed**
* `POST` a question using a `question_id`

**How can this be manually tested?**
* Clone this repo and install all dependencies
* run `export FLASK_CONFIG='development'`
* Run app 
* open postman and enter `/api/v1/questions/<int:question_id>/answers` using a  `POST` method
* replace `<int:question_id>` with the question id you want to post an answer for in `int` format
* Post the answer

**What are the relevant PT stories?**
https://www.pivotaltracker.com/story/show/160215770

### CODE REVIEW

File: API/app/answers/routes.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+"""
+from flask import jsonify, request, abort
+
+from app.answers import BP
+from models.models import Answers, Questions
+
+
+
+@BP.route('/api/v1/questions/<int:question_id>/answers', methods=['POST'])
+def post_a_answers(question_id):
```
Comment (graycadeau) [inline API/app/answers/routes.py]: Did you mean `post_answers` ?

File: API/app/answers/routes.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    """
+    Post an answer to a question endpoint
+
+    :return: "json object"
+    :raises Exeption: if there is an error in `JSON` parsing
+    """
+    if not request.json or not 'answer' in request.json:
+        abort(400)
```
Comment (graycadeau) [inline API/app/answers/routes.py]: I think you could use an error message for that

File: API/app/answers/routes.py (hunk 1 above)
Comment (maxwellgithinji) [reply API/app/answers/routes.py]: noted thanks

File: API/app/answers/routes.py (hunk 2 above)
Comment (maxwellgithinji) [reply API/app/answers/routes.py]: i will do that. thanks

File: API/app/answers/routes.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        return jsonify({'error':"Invalid request"}), 400
+
+    my_answer = {
+        'id': request.json.get('id', ""),
+        "answer": request.json['answer'],
+        "question_id": request.json.get('question_id', ""),
+        "author_id": 1
+    }
+
+    with request:
```
Comment (Dhul-Husni) [inline API/app/answers/routes.py]: why with request

File: API/app/answers/routes.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        if my_answer['answer'].strip() == "":
+            return jsonify({'error':"Cannot Post Blank Answer!"}), 400
+        if my_answer['question_id'] is None:
```
Comment (Dhul-Husni) [inline API/app/answers/routes.py]: `if not my_answer['question_id']` is more elegant

File: API/app/answers/routes.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            return jsonify({'error':"Cannot Post Blank question_id!"}), 400
+        if not isinstance(my_answer['question_id'], int):
+            return jsonify({'error':"Only Integers Allowed"}), 400
+
+    answer = [
+        current_answer for current_answer in Answers
```
Comment (Dhul-Husni) [inline API/app/answers/routes.py]: :\

Comment (Dhul-Husni) [reply API/app/answers/routes.py]: still looping

File: API/app/answers/routes.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        if current_answer['answer'] == my_answer['answer']
+        ]
+
+    my_question = [
+        current_question for current_question in Questions
```
Comment (Dhul-Husni) [inline API/app/answers/routes.py]: As is here :/

File: API/app/questions/routes.py [hunk 7]
```diff
@@ -17,9 +17,9 @@ def question_endpoint():
     :raises Exeption: if there is an error in `JSON` parsing
     """
     try:
-        if not QUESTIONS:
+        if not Questions:
```
Comment (Dhul-Husni) [inline API/app/questions/routes.py]: why do your variables start with a caps
i thing both Answers and Questions should be `questions` and `answers`

**Pull request merged**